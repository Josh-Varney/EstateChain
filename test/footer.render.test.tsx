import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import LandingSubscription from "../src/components/landing/components/footer/footer";

// Mock child components
jest.mock("../src/components/landing/components/footer/components/footer-form", () => () => (
    <div data-testid="subscription-form">Subscription Form</div>
));
jest.mock("../src/components/landing/components/footer/components/footer-card", () => () => (
    <div data-testid="links-card">Links Card</div>
));
jest.mock("../src/components/landing/components/footer/components/footer-info", () => () => (
    <div data-testid="footer">Footer</div>
));

describe("LandingSubscription Component", () => {
    it("should render the SubscriptionForm component", () => {
        render(<LandingSubscription />);
        expect(screen.getByTestId("subscription-form")).toBeInTheDocument();
    });

    it("should render the LinksCard component", () => {
        render(<LandingSubscription />);
        expect(screen.getByTestId("links-card")).toBeInTheDocument();
    });

    it("should render the Footer component", () => {
        render(<LandingSubscription />);
        expect(screen.getByTestId("footer")).toBeInTheDocument();
    });

    it("should match the snapshot", () => {
        const { container } = render(<LandingSubscription />);
        expect(container).toMatchSnapshot();
    });
});
