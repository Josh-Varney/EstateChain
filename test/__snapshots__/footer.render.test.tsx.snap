// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingSubscription Component should match the snapshot 1`] = `
<div>
  <div
    class="w-full px-4 sm:px-8 lg:px-12 overflow-hidden"
  >
    <div
      class="grid grid-cols-1 md:grid-cols-12 gap-6"
    >
      <div
        data-testid="subscription-form"
      >
        Subscription Form
      </div>
      <div
        data-testid="links-card"
      >
        Links Card
      </div>
    </div>
    <div
      data-testid="footer"
    >
      Footer
    </div>
  </div>
</div>
`;
